usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java

```diff
@@ -0,0 +1,216 @@
+        if (isNegative(year, month, hour, minutes, seconds) || isOutOfRange(month, hour, minutes, seconds)) {
+            throw new InvalidFormatException();
+        }
+    }
+
+
+    /**
+     * Method responsible for checking whether the date values are out of range or not.
+     * @param month The month value that we want to check
+     * @param hour The hour value that we want to check
+     * @param minutes The minutes value that we want to check
+     * @param seconds The seconds value that we want to check
+     * @return True if any value is out of range or false otherwise
+     */
+    private boolean isOutOfRange(int month, int hour, int minutes, int seconds) {
+        int nDays = numberOfDays(year, month);
+        boolean dateOutOfRange = month > DECEMBER || day > nDays;
+        boolean hourOutOfRange = hour >= MAX_HOUR || minutes >= MAX_MINUTES_SECONDS || seconds > MAX_MINUTES_SECONDS;
+        return dateOutOfRange || hourOutOfRange;
+    }
+
+    /**
+     * Method responsible for checking whether the date values are negative or not.
+     * @param year The year value that we want to check
+     * @param month The month value that we want to check
+     * @param hour The hour value that we want to check
+     * @param minutes The minutes value that we want to check
+     * @param seconds The seconds value that we want to check
+     * @return True if any value is negative or false otherwise
+     */
+    private boolean isNegative(int year, int month, int hour, int minutes, int seconds) {
+        boolean isYearNegative = year < 0 || month < 0 || day < 0;
+        boolean isDateNegative = hour < 0 || minutes < 0 || seconds < 0;
+        return isYearNegative || isDateNegative;
+    }
+
+    /**
+     * Method responsible for obtaining the number of days for a given month of a given year.
+     * @param year The year value
+     * @param month The month for which we want to obtain the number of days
+     * @return The number of days of the month for that given year
+     */
+    private int numberOfDays(int year, int month) {
+        if (month <= JULY) {
+            if (month != FEBRUARY) {
+                if (month % 2 == 0) {
+                    return DAYS_30;
+                } else {
+                    return DAYS_31;
+                }
+            } else if (isLeapYear(year)) {
+                return DAYS_29;
+            } else {
+                return DAYS_28;
+            }
+        } else if (month % 2 == 0) {
+            return DAYS_31;
+        }
+        return DAYS_30;
+    }
+
+    /**
+     * Method responsible for checking whether a year is a leap year or not.
+     * @param year The year that we want to check
+     * @return True if the given year is a leap year or false otherwise
+     */
+    private boolean isLeapYear(int year) {
+        return year % LEAP_YEAR_FREQ == 0 && (year % TWO_LAST_DIGITS != 0
+            || (year / TWO_LAST_DIGITS) % LEAP_YEAR_FREQ == 0);
+    }
+
+    @Override
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [NCSS for this method is 19 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)